kubernetes/template-generator/cli.py

```diff
@@ -0,0 +1,143 @@
+        mandatory_groups.extend([mandatory])
+
+        #selector = p.add_argument_group("Template selectors (optional)")
+        #selector.add_argument("--pod", "-p", action="store_true", default=None, help="Generate pod template (default: true)")
+        #selector.add_argument("--service", "-s", action="store_true", default=None, help="Generate service template (default: true)")
+        #selectors.extend([selector])
+
+
+    #multivolume specific
+    mandatory_groups[0].add_argument("--image-url", "-m", help="URL to machine tar image", required=True)
+
+    #macro specific
+    #selectors[1].add_argument("--docker", "-d", action="store_true", default=None, help="Generate dockerfile (default: true)")
+    macro.add_argument("--local-image", "-l", action="store_true", help="Disable pulling images on node (Default: false)", default=False)
+
+    return parser.parse_args()
+
+
+def write_file(path, content):
+    f = open(path, "w")
+    f.write(content)
+    f.close()
+
+def main():
+    params = parse_params()
+
+    path = "{}/{}".format(LEAPP_BASEPATH, params.container_name)
+
+    ## Find release version
+    rfile = open("{}/etc/redhat-release".format(path), "r")
+    ver = re.findall(".*\srelease\s(\d+)(?:\.\d+)?.*", rfile.readline(64))
+    rfile.close()
+
+    if len(ver) and ver[0] == "6":
+        system = RHEL6ContainerOS()
+    elif len(ver) and ver[0] == "7":
+        system = RHEL7ContainerOS()
+    else:
+        raise Exception("Unknown release version")
+
+    sanit_name = TemplateGenerator.sanitize_container_name(params.container_name)
+    dest = params.dest
+
+    print("Storing templates to: {}".format(dest))
+
+    if params.generator == "macroimage":
+        #selective_templating = False
+
+        #for i in params.pod, params.service, params.docker:
+        #    if i:
+        #        selective_templating = True
+
+        template = MacroimageTemplateGenerator(
+            params.container_name,
+            system,
+            exposed_ports = params.tcp,
+            is_local = params.local_image,
+            external_ips = params.ip
+            )
+
+
+
+        write_file("{}/{}-svc.yaml".format(dest, sanit_name), template.generate_service_template())
+        write_file("{}/{}-pod.yaml".format(dest, sanit_name), template.generate_pod_template())
+        write_file("{}/Dockerfile".format(dest, sanit_name), template.generate_dockerfile())
+
+
+        print("Create a tar image of migrated machine:")
+        print("  tar -czf {}/{}.tar.gz -C {} .\n".format(dest, sanit_name, path))
+        print("Move it to the machine where you want to build the container \n\
+along with generated Dockerfile and execute: ")
+        print("  docker build -t {} .\n".format(template.macroimage_name()))
+
+        if not params.local_image:
+            print("Push the new image into registry")
+            print("  docker push {}".format(template.macroimage_name()))
+
+
+    elif params.generator == "multivolume":
```

Comment:
i think the entire content of this if and elif could be placed in 2 different functions to avoid this huge if blocks